src/api/vdom.js

```diff
@@ -166,17 +166,25 @@ function wrapIdomFunc(func, tnameFuncHandler = () => {}) {
       const isElementClosing = func === elementClose;
       const isElementOpening = func === elementOpen;
 
-      // If we're skipping the tree, we must skip everything except for the
-      // closing of the element that originally started the skipping.
-      if (skipCurrentTree && !isElementClosing && !currentElement()[$skipCurrentElement]) {
-        return;
+      if (isElementOpening) {
+        if (skipCurrentTree) {
+          return;
+        } else {
+          const elem = func(...args);
+          skipCurrentTree = !!elem[$skipCurrentElement];
+          return elem;
+        }
       }
+      
+      if (isElementClosing) {
+        let elem = currentElement();
 
-      const elem = func(...args);
+        if (elem[$skipCurrentElement]) {
+          skipCurrentTree = false;
```

Comment:
Shouldn't we call [`skipNode`](https://github.com/google/incremental-dom/blob/master/src/core.js#L452) here too?